### PR TITLE
Add S1105 (based on `brace-style`) for JS/TS

### LINKED ISSUE #1404: Add S1105 (based on `brace-style`) for JS/TS
`brace-style` also reports on closing brace before else, we can extend our rule (update JS RSPEC)

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**
Fixes #1404

### CODE REVIEW

**Comment by quentin-jaquier-sonarsource** [comment]
### Changes in the rule
- **new issues** when the closing brace is not on the same line as the `else`, `if-else`, `catch` or `finally keyword`.
- **new issues** when a statements inside a brace appears at the same line as the opening or closing brace.

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S1105.html [hunk 1]
```diff
@@ -1,25 +1,48 @@
-<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when an open curly brace is not placed at the end of a
-line of code.</p>
+<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when:</p>
+<p> * an open curly brace is not placed at the end of a line of code.</p>
+<p> * the closing brace is not on the same line as the <code>else</code>, <code>if-else</code>, <code>catch</code> or <code>finally</code>
+keyword.</p>
+<p> * a statements inside brace appears at the same line as the opening or closing brace.</p>
```
Comment (vilchik-elena) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S1105.html]: remove `s` from statement

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S1105.html [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
 <pre>
 if (condition)
 {                                                      //Noncompliant
   doSomething();
+}                                                      //Noncompliant
+else {
+  doSomethingElse();
 }
+
+try { somethingRisky();                            //Noncompliant
```
Comment (vilchik-elena) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S1105.html]: do we really need example with `try`?

Comment (quentin-jaquier-sonarsource) [reply javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S1105.html]: I added this one to illustrate an issue for the third case: 

> a statement inside brace appears at the same line as the opening or closing brace.

Comment (quentin-jaquier-sonarsource) [reply javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S1105.html]: Anyway, I change the description and there is no more need for it.

**Comment by vilchik-elena** [comment]
@quentin-jaquier-sonarsource update ticket that we've added a parameter to the rule

**Pull request merged**